### PR TITLE
Various headers from public Leptos and Apple projects

### PR DESCRIPTION
**Pull request by leptos-null**
Sources:

[GeoServices](https://github.com/leptos-null/geoservices-example/tree/f7d3ab5812d8106a7d893780fda39443d1087561/GeoServices)

[PersonaKit](https://github.com/leptos-null/Persona-Example/tree/3078a8889ed5de914c1dfa62456672be00981b8e/PersonaKit)

[PersonUI](https://github.com/leptos-null/Persona-Example/tree/3078a8889ed5de914c1dfa62456672be00981b8e/PersonaUI)

[MediaPlayer](https://github.com/WebKit/webkit/blob/a5a1e58afc73d2aecc4f47eee86399c9bc5622ea/Source/WebCore/PAL/pal/spi/ios/MediaPlayerSPI.h)

[Celestial](https://github.com/WebKit/webkit/blob/90fb821b0ea1ec5b37a68ff01e6e539afcbae92c/Source/WebCore/PAL/pal/spi/ios/CelestialSPI.h)

### CODE REVIEW

**Comment by leptos-null** [comment]
Second round of commits are from my private projects. 

In dbd426c6b652ea25fa6b3a3eb1f88874c1151ef2, I removed `-[UIStatusBarItem setView:forManager]` (added in b6f4294956f93cf83ae2bc63c5bf1ec09c4f6d60 in 2015) because it's never existed as far as I could find (I did my best to check from iOS 4 to iOS 11)

File: UIKit/UIStatusBarItem.h [hunk 1]
```diff
@@ -1,17 +1,62 @@
...
 @property (nonatomic, readonly) NSString *indicatorName;
+@property (nonatomic, readonly) UIStatusBarItemType type;
+@property (nonatomic, readonly) Class viewClass; // a subclass of UIStatusBarItemView
 
-- (instancetype)initWithType:(UIStatusBarItemType)type;
+@property (nonatomic, readonly) int priority;
+@property (nonatomic, readonly) int leftOrder;
+@property (nonatomic, readonly) int rightOrder;
 
-- (void)setView:(UIStatusBarItemView *)view forManager:(UIStatusBarLayoutManager *)layoutManager;
```
Comment (leptos-null) [inline UIKit/UIStatusBarItem.h]: I can't find evidence that this ever existed

**Comment by leptos-null** [comment]
BatteryCenter headers are from my [Batsie](https://github.com/leptos-null/Batsie/tree/9cb8b4cbb07d8d4e3b5cf4d5a459e5a5ee1b981a/BatteryCenter) project.

File: UIKit/UIStatusBarItem.h (hunk 1 above)
Comment (kirb) [reply UIKit/UIStatusBarItem.h]: May have been mixed up with some libstatusbar stuff. Agreed, best to remove it.

Comment (kirb) [reply UIKit/UIStatusBarItem.h]: Confirmed: https://github.com/phoenix3200/libstatusbar/blob/dc5877e41891ed57a08593ab2e26fd3e9818bd2b/UIStatusBarCustomItem.mm#L174

**Comment by kirb** [comment]
This looks really good!

* Could you make sure properties are using `@property` rather than defining their get/set methods?
* Check where `int`, `float`, etc are used and consider if they should be `NSInteger`, `CGFloat`, or some other type more precise to what the value represents.

**Comment by leptos-null** [comment]
Thanks!

- The places were I saw get/set methods are in [BluetoothDevice](https://github.com/nst/iOS-Runtime-Headers/blob/fbb634c78269b0169efdead80955ba64eaaa2f21/PrivateFrameworks/BluetoothManager.framework/BluetoothDevice.h), [BluetoothManager](https://github.com/nst/iOS-Runtime-Headers/blob/fbb634c78269b0169efdead80955ba64eaaa2f21/PrivateFrameworks/BluetoothManager.framework/BluetoothManager.h), and VolumeControl. These instances don't have corresponding properties in the runtime. I can replace these seemingly getters and setters with properties that would compile to the same thing, but it wouldn't reflect the runtime information (I don't think it matters if it was declared as a property here, and not originally, so I'll do that if you'd like)
    - Unimportant, but my personal analysis is that given these classes don't have class prefixes (i.e. BluetoothManager classes should have a class prefix of `BM` or similar), and that `BluetoothDevice` and `BluetoothManager` are entirely wrappers for C structs (`BTDeviceImpl ` and `BTAccessoryManagerImpl` respectively), these classes were likely authored by less skilled users, and properties were indeed never used 
- The two main places that I'm seeing `int`s and `float`s are, again, in the BluetoothManager framework, and in Celestial framework and VolumeControl, respectively. As I mentioned previously, BluetoothManager is a wrapper for the lower-level structs, which seem to match up with the Bluetooth specification (more [research](https://www.bluetooth.org/docman/handlers/downloaddoc.ashx?doc_id=457080) required), and those types are highly likely to be `int32_t`s and `uint32_t` for the identifiers, however I wasn't confident enough to mark them as such. Celestial and VolumeControl both deal with volume, which is handled as a `float` on iOS. The headers I based these on were dumped from a 64-bit runtime, so I would expect to see `long long`s and `double`s. The place were I see `double`s are in the `GeoServices` code- on a 32-bit target, these are still `double`s. I don't see any `long long`s or other "basic" primitives.

I understand your comments, and I don't mean to criticize them at all. I've written a lengthy response in an attempt to show my reasoning to anyone who wishes to consume these resources, and more importantly, in the hopes that another my find a flaw in my reasoning, resulting in better resources

**Comment by kirb** [comment]
Alright that makes sense, I just wanted to be sure you didn't miss something. A property would make more sense for readability since that's what these headers are about, but if it definitely isn't written using properties in the first place, then I think we should be accurate to that.

I'll merge this as it is then. Thanks!

**Comment by leptos-null** [comment]
Thanks! I agree

**Comment by kirb** [comment]
Merged in dc399ef

**Pull request merged**